backend/users/models/token.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+from datetime import datetime
+from datetime import timedelta
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from rest_framework.authtoken.models import Token
+
+
+class BearerToken(Token):
+    note = models.TextField(null=True)
+    expires_at = models.DateTimeField(null=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL, related_name="bearer_tokens", on_delete=models.CASCADE, verbose_name=("Users")
+    )
+    id = models.UUIDField(default=uuid.uuid4, editable=False)
+
+    @property
+    def is_expired(self) -> bool:
+        if self.expires_at is None:
+            return False
+        return self.expires_at < timezone.now()
+
+
+class ImplicitBearerToken(Token):
+    """
+    Legacy token to make the endpoint api-token-auth/ work like it used to
+    """
+
+    @property
+    def expires_at(self) -> datetime:
+        return timezone.now() + self.time_left
```

Comment:
I feel like using `time_left` there makes useless calculation. This equals to:
```
timezone.now() + self.time_left
-> timezone.now() + (settings.EXPIRY_TOKEN_LIFETIME - time_elapsed)
  -> timezone.now() + (settings.EXPIRY_TOKEN_LIFETIME - (timezone.now() - self.created))
    -> timezone.now() + (settings.EXPIRY_TOKEN_LIFETIME - timezone.now() + self.created)
      -> settings.EXPIRY_TOKEN_LIFETIME + self.created
```
We could simplify this line to `settings.EXPIRY_TOKEN_LIFETIME + self.created`. I'm also not sure that the two `timezone.now()` are going to be equal.

Then, as it looks like that `time_left` is only used there, then I would then rather just replace `is_expired` with the line `return self.expires_at > timezone.now()`  and remove `time_left`